### ISSUE TITLE
Theme: Custom Commands columns are gone

### ISSUE THREAD

**Issue by strangealien** [authentic-theme/authentic-theme#1245]
I had a carefully ordered Custom Commands page in Webmin, with 8 different commands and descriptions (short descriptions, about half a line) in a table with groups of 4 commands in two columns.

After the last update, this degenerated into an ugly column of 8 commands in wrong order...
Argh.

It ignores the setting to order it in two columns, it just shows them in one column only, I can order the commands with their names, then a resemblance of order returns, but I liked _my_ order, not some alphabetical order I did not want or need!
Also, the single column has the buttons for the commands left, then the descriptions, and then a curious rectangle on the right border. After changing my browsers window size to very small, and back, the small rectangles become 'edit command' buttons. They change back fast to rectangles after this, they never remain buttons with text inside. But in link mode they remain buttons with run/edit.

I want the old look and order back, please! I had a carefully ordered two column view with the command buttons/links and the description below. I want this back please, the new thing is not an improvement at all! Especially I want two columns, because it was planned like this!!

I tried to change from buttons to links hoping this would bring back my command table, but no luck. Actually I tried all settings for this page, except the editor settings, nothing changed it back, it is always one ugly column of 8 commands, with either rectangles on the right (the 'edit buttons') or in the link mode with actual edit buttons. 

Using Firefox btw.
Thank you.

**Comment by rostovtsev** [comment]
Hi,

I am sorry that you don't like new look of Custom Commands. Let's work it out and improve it!

I must say, that ordering works correctly with new output. 

There are _no_ two columns at the moment. I missed _Columns to display commands in_ option to be respected.

At first here is how it looks for me in different themes using Firefox:

![image](https://user-images.githubusercontent.com/4426533/48934563-9a7c7700-ef15-11e8-847e-995a7da9e3d9.png)

How does it look for you?

I will try to take 2 columns output back.

**Comment by strangealien** [comment]
Hello and thank you for looking into this. And sorry for the title - I was a bit miffed when I wrote it.
Before it looked exactly like your picture - the right one. Now it looks exactly like your picture - the left one.
There is one difference: Using the size of my FF window (I use it quite big, since I have a 2560x1600 monitor) the buttons on the right in the new view are just rectangles, if I drag the right border of the Firefox window and make it smaller, the 'edit' buttons appear, but if I change into another tab and back, the rectangles are back again.
I tried to attach a picture from it.

**Comment by strangealien** [comment]
Hello,
and thank you for looking into this issue. Also - sorry for the title, I was a bit miffed when I wrote it. No offense meant, I just had a mixup with the latest systemd update, which botched the Webmin restart after the update...
My Custom Commands Window looks like your picture above, the left version, and before it looked like your picture, the right version. The only difference is, that the buttons (your picture in the left) are not buttons with text in it, they are just small rectangles. If I drag the right window border from Firefox to make it smaller, the buttons appear, and remain until I change into another tab, when I go back again, the rectangles are back, and I have to adjust FF size to get the buttons with text. But the rectangles seems to work too.
![window](https://user-images.githubusercontent.com/37230572/48940046-31453580-ef0e-11e8-83a3-3e8485e2bdee.jpg)

I am using an ESR version of Firefox, since the new versions support only one of my 5 Add Ons, and it is butt ugly too, the interface nearly unusable. If one the newest version of Firefox is supported I will have to change back to the old Webmin interface. I will not use the new Firefox, and spend hours after every update to adjust the interface to an usable state. But this rant does not belong here :)
Thank you

**Comment by rostovtsev** [comment]
Okay,

The buttons in your Firefox ESR are not shown? What version of Firefox ESR? Is it version 60?

Did you try to use Firefox Private Window (where all addons are disabled)? I'm surprised that there is no text in buttons. I meant buttons' text to be _always_ there.

Let's summarize - having buttons' text to be there (intended initially) and having two column output work (missing now) would be the right improvement?

**Comment by rostovtsev** [comment]
I just tried Firefox ESR 60 without any extensions and it shows the text in the buttons on the right, like _Edit Command_, _Edit File Editor_ and etc.

**Comment by strangealien** [comment]
I am using FF 52.9.0 (32 bit), since the update to 56.x destroyed my database and disabled 3 of my 4 Add Ons. I am still looking for an alternative. So - if these buttons do not work an a Firefox version older than 56.x, it is okay with me. I am still thinking about Waterfox and other alternatives, but Mozilla has proven that they do not care about many users. Disabling more than 70% of all Add Ons with their changes, not to mention the awful new flat UI look made me look for alternatives. But this is not Webmins problem :)
So - if it does not work because I am using an ESR version, which is outdated now for around one  month, it is okay with me - and sorry for the troubles.
But having two columns would really help here.
Thank you

**Comment by rostovtsev** [comment]
Do you have any errors in browser's console when running index page of Custom Commands, where buttons don't show the text?

**Comment by rostovtsev** [comment]
I have fixed the output as two columns. You are right - it's important and now, even nice looking feature.

I have installed _Firefox ESR 52.9_ for you to double check on buttons - all works fine. I can see the text on the buttons as expected. I think the issue is cause by one of the extensions.

You are welcome to give the latest patch a try, simply by going to theme configuration and installing latest development version of the theme.

Enjoy. :smiley_cat:

**Comment by strangealien** [comment]
Hello and Thank you!
I just saw your request for errors in the browser console. I have no experience with this (not really), but there is one error shown as soon as I open Webmin on my server, even without logging in:
Error reading value for 'max-height'. Declaration ignored.
(Translation from german original: 'Fehler beim Verarbeiten des Wertes für 'max-height'.  Deklaration ignoriert.')
This error repeats, but I have no idea what it means. When I did (very beginner level) css for my web site, I sometimes saw similar errors. No idea if it is serious.
Will check latest patch now and report ASAP.
Thank you!
